File: functions/src/types/calendar.ts
// Calendar-specific types and interfaces

export interface CalendarEvent {
  id: string;
  summary: string;
  start: {
    dateTime?: string | null;
    date?: string | null;
  };
  end: {
    dateTime?: string | null;
    date?: string | null;
  };
  location?: string | null;
  description?: string | null;
}

export interface CalendarRequest {
  accessToken: string;
  calendarId?: string;
  timeMin?: string;
  timeMax?: string;
  maxResults?: number;
}

export interface CalendarEventsRequest {
  timeMin?: string;
  timeMax?: string;
  maxResults?: number;
  calendarId?: string;
}

export interface CalendarAuthRequest {
  googleToken: {
    access_token: string;
    refresh_token?: string;
    scope: string;
    token_type?: string;
    expiry_date?: number;
  };
}

export interface CalendarStatusResponse {
  hasAccess: boolean;
  error?: string;
}

export interface CalendarEventsResponse {
  success: boolean;
  events: CalendarEvent[];
  count: number;
  error?: string;
}

File: backend/internal/model/calendar.go
package model

import "time"

type CalendarEvent struct {
	ID          string    `json:"id"`
	Summary     string    `json:"summary"`
	Description string    `json:"description"`
	Location    string    `json:"location"`
	Start       time.Time `json:"start"`
	End         time.Time `json:"end"`
	AllDay      bool      `json:"all_day"`
	Attendees   []string  `json:"attendees,omitempty"`
}

type CalendarSyncRequest struct {
	TimeMin    time.Time `json:"time_min"`
	TimeMax    time.Time `json:"time_max"`
	MaxResults int       `json:"max_results"`
}

type CalendarSyncResponse struct {
	Success   bool            `json:"success"`
	Events    []CalendarEvent `json:"events,omitempty"`
	Total     int             `json:"total"`
	SyncedAt  time.Time       `json:"synced_at"`
	Error     string          `json:"error,omitempty"`
}

